xds/internal/xdsclient/xdslbregistry/converter/converter.go

```diff
@@ -34,81 +34,33 @@ import (
 	internalserviceconfig "google.golang.org/grpc/internal/serviceconfig"
 	"google.golang.org/grpc/xds/internal/balancer/ringhash"
 	"google.golang.org/grpc/xds/internal/balancer/wrrlocality"
+	"google.golang.org/grpc/xds/internal/xdsclient/xdslbregistry"
 
 	v1xdsudpatypepb "github.com/cncf/xds/go/udpa/type/v1"
 	v3xdsxdstypepb "github.com/cncf/xds/go/xds/type/v3"
-	v3clusterpb "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
 	v3clientsideweightedroundrobinpb "github.com/envoyproxy/go-control-plane/envoy/extensions/load_balancing_policies/client_side_weighted_round_robin/v3"
 	v3ringhashpb "github.com/envoyproxy/go-control-plane/envoy/extensions/load_balancing_policies/ring_hash/v3"
 	v3wrrlocalitypb "github.com/envoyproxy/go-control-plane/envoy/extensions/load_balancing_policies/wrr_locality/v3"
 	structpb "github.com/golang/protobuf/ptypes/struct"
 )
 
-var (
-	// m is a map from proto type to converter.
-	m = make(map[string]converter)
-)
-
 func init() {
 	// Construct map here to avoid an initialization cycle.
-	m = map[string]converter{
+	xdslbregistry.SetRegistry(map[string]xdslbregistry.Converter{
```

Comment:
IMO this now would look better as individual calls to register instead of a full initializion.  That way we could add implementations in other places if we wanted.

Reply:
Switched.